tl-services/src/main/java/org/egov/tl/validator/TLValidator.java

```diff
@@ -215,6 +217,62 @@ private void validateInstitution(TradeLicenseRequest request){
     }
 
 
+
+    /**
+     *  Validates the fromDate and toDate of the request
+     * @param request The input TradeLicenseRequest Object
+     */
+    public void validateRenewal(TradeLicenseRequest request){
+        Map<String,String> errorMap = new HashMap<>();
+        
+        TradeLicenseSearchCriteria criteria = new TradeLicenseSearchCriteria();
+        List<String> ids = new LinkedList<>();
+        List<String> licenseNumbers = new LinkedList<>();
+        request.getLicenses().forEach(license -> {
+            ids.add(license.getId());
+            if(license.getLicenseNumber() != null){
+                licenseNumbers.add(license.getLicenseNumber());
+            }            
+        });
+        criteria.setTenantId(request.getLicenses().get(0).getTenantId());
+        criteria.setIds(ids);
```

Comment:
New ids are already enriched by this point it won't return you any license. Also there should be no if condition on license Number if it's not there we should throw error. We should instead set APPROVED status in the search criteria with tenantId and licenseNumbers